### PR TITLE
Replaced id with nqn in subsystem requests and added result status

### PR DESCRIPTION
**Pull request by amarv-marvell**
Subsystem is uniquely identified by nqn. By replacing id with nqn, we can save one level of indirection.

Signed-off-by: Amar Vishwakarma <amarv@marvell.com>, Michal Kalderon <mkalderon@marvell.com>

### CODE REVIEW

File: storage/proto/frontend.proto [hunk 1]
```diff
@@ -125,23 +124,23 @@ message NVMeSubsystemCreateRequest {
 }
 
 message NVMeSubsystemCreateResponse {
-    // Intentionally empty.
+    uint32 result = 1;
```
Comment (glimchb) [inline storage/proto/frontend.proto:127]: I was thinking as a result of every `create` we can return UUID, wdyt ?

File: storage/proto/frontend.proto [hunk 2]
```diff
@@ -67,7 +67,6 @@ service VirtioScsiLunService {
 // TODO: Virtio-fs
 
 message NVMeSubsystem {
-    int64 id = 1;
```
Comment (glimchb) [inline storage/proto/frontend.proto:70]: need to adjust fields numbering ?

**Review by glimchb — COMMENTED** [review]
thanks, looks good, few small comments inside

**Comment by glimchb** [comment]
@amarv-marvell please also run [Makefile](https://github.com/opiproject/opi-api/blob/main/storage/Makefile) 
it will auto-generate new documentation

File: storage/proto/frontend.proto (hunk 1 above)
Comment (amarv-marvell) [reply storage/proto/frontend.proto:127]: I think the result should be error/status codes to be uniformly applicable to all services.
As per the spec, NQN themselves sometimes can consist of UUIDs. But whether it consists UUID or not,
it is always unique, so I think another UUID may not be needed.

**Comment by glimchb** [comment]
@amarv-marvell the PR was auto-closed, it is is a good start, did you intended to close it ? you need any help ?

**Comment by amarv-marvell** [comment]
> @amarv-marvell the PR was auto-closed, it is is a good start, did you intended to close it ? you need any help ?

It was closed by mistake since I sync'ed main repo back to my fork which caused an unnecessary merge commit in PR, while resetting the same PR got closed. I intend to reopen it.
I think a new commit might reopen it, or you could help.
Also I ran make as suggested by you and saw that my docker by default picks  protoc-gen-go v1.28.1 instead of v1.28.0
Due to this I see quite a lot of changes in generated .pb.go files
Any clue on how to make docker pick protoc-gen-go v1.28.0 would be great.

**Comment by glimchb** [comment]
@mestery any idea why DCO isn't running ?

**Comment by amarv-marvell** [comment]
> @mestery any idea why DCO isn't running ?

Looks like its hung. I will retry the push after making the change of returning uuid for NVMeSubsystemCreateResponse as both you and @mestery recommend.

File: storage/proto/frontend.proto (hunk 1 above)
Comment (amarv-marvell) [reply storage/proto/frontend.proto]: Made the change to return uuid in NVMeSubsystemCreateResponse.

**Comment by amarv-marvell** [comment]
> > @mestery any idea why DCO isn't running ?
> 
> Looks like its hung. I will retry the push after making the change of returning uuid for NVMeSubsystemCreateResponse as both you and @mestery recommend. New push should retrigger DCO.

It has passed this time!

**Review by glimchb — APPROVED** [review]
thanks @amarv-marvell for addressing the comments
please create new PR to this repo https://github.com/opiproject/opi-poc/tree/main/storage
using new changed fields, otherwise it will become broken

**Comment by amarv-marvell** [comment]
> thanks @amarv-marvell for addressing the comments please create new PR to this repo https://github.com/opiproject/opi-poc/tree/main/storage using new changed fields, otherwise it will become broken

will do

**Comment by glimchb** [comment]
fyi @GottliebNoam

**Pull request merged**